### ISSUE TITLE
Fails to report stats for PRs from a fork

### ISSUE THREAD

**Issue by developit** [preactjs/compressed-size-action#2]
Currently certain PR's coming from forks result in the action running in the context of the fork, which means it can't post a comment or a PR review to the PR/issue.

Possible workarounds:

1. Report the size table back using a [Status Check](https://github.com/qawolf/actions-test/blob/master/.github/actions/create-status-check/index.js#L15-L27)
    - Not sure if GitHub's UI will handle this
    - Not sure if status checks have the same permissions issue
    - Not sure if status checks for the HEAD or merge commit on a PR show on the PR itself

**Comment by kuldeepkeshwar** [comment]
@developit  In case of forks, the passed `GITHUB_TOKEN` has read-only permission (due to security constraints), even if we go for `status check`, the token will still have read-only permission and won't be able to update upstream. 

The ideal solution for such case is Github App(I was hacking around the same problem while creating [Size plugin bot](https://github.com/kuldeepkeshwar/size-plugin-bot), my initial idea was to go for Github Action, but couldn't find any workaround )

**Comment by developit** [comment]
Yeah, it's not great that actions have this limitation. I had actually tried doing a `status check` version of this and came to the same conclusion.

**Comment by ZLY201** [comment]
Maybe change this action to `github app`?

Just like [coveralls/coveralls](https://github.com/marketplace/actions/coveralls-github-action), it has a github robot named [`coveralls`](https://github.com/coveralls).

When you run coveralls the robot will comment in the PR.

We can even invite the robot as a outside collaborator to ensure it has the comment permission.

**Comment by ZLY201** [comment]
> @ZLY201 GitHub robot is already there [size-plugin-bot](https://github.com/kuldeepkeshwar/size-plugin-bot)

Thanks! I'll try it!

**Comment by rzzf** [comment]
If Markdown content can be exposed via outputs and the user manages the comment publishing themselves, then using workflow_run to post comments and running the action in pull_request should solve the problem, right?

**Comment by rschristian** [comment]
> If Markdown content can be exposed via outputs and the user manages the comment publishing themselves, then using workflow_run to post comments and running the action in pull_request should solve the problem, right?

I'm not sure about outputs specifically, but you certainly can do something like that, yes. I've done it in the past with artifacts.

It's a two-step process running from entirely different contexts is all.

**Comment by rzzf** [comment]
> > If Markdown content can be exposed via outputs and the user manages the comment publishing themselves, then using workflow_run to post comments and running the action in pull_request should solve the problem, right?
> 
> I'm not sure about outputs specifically, but you certainly can do something like that, yes. I've done it in the past with artifacts.
> 
> It's a two-step process running from entirely different contexts is all.

Thank you for such a quick response! ❤️ 

I’ve tried something similar before, 

- https://github.com/element-plus/element-plus/blob/dev/.github/workflows/lint-commit-message.yml
- https://github.com/element-plus/element-plus/blob/dev/.github/workflows/lint-commit-message-post.yml

but it seems that `compressed-size-action` currently doesn’t expose the results, unless we parse the workflow logs, which feels a bit hacky. 
Maybe we could support this use case? For example, allowing the built-in comment to be disabled and exposing the results for developers to handle themselves?

**Comment by rschristian** [comment]
> Maybe we could support this use case? For example, allowing the built-in comment to be disabled and exposing the results for developers to handle themselves?

Sounds like a very good idea, yep. If outputs work, I'd say we can always pipe the message to the output AND try to comment, just in case someone wants both behaviors and step outputs (to my knowledge) are largely unproblematic to add just in case.

PRs welcome.

**Comment by rzzf** [comment]
> > Maybe we could support this use case? For example, allowing the built-in comment to be disabled and exposing the results for developers to handle themselves?
> 
> Sounds like a very good idea, yep. If outputs work, I'd say we can always pipe the message to the output AND try to comment, just in case someone wants both behaviors and step outputs (to my knowledge) are largely unproblematic to add just in case.
> 
> PRs welcome.

@rschristian I created a [PR](https://github.com/preactjs/compressed-size-action/pull/148) to try to address this issue. Please take a look when you have time, thank you!

**Comment by rzzf** [comment]
I think this issue can be closed now. I've successfully implemented and validated this in a real-world project.  https://github.com/element-plus/element-plus/pull/24440

Resolve by #148

**Comment by rschristian** [comment]
IMO, we will close this if and when we offer a solution here ourselves, such as a second action or official example to use. Don't really want to make users reference a third party action (which may change over time) that lacks comments on why certain things are done.
